File: Java/src/test/java/org/codecop/lcdnumbers/LineTest.java
package org.codecop.lcdnumbers;

import nl.jqno.equalsverifier.EqualsVerifier;
import org.junit.Rule;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnit;
import org.mockito.junit.MockitoRule;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.when;

public class LineTest {

    final String left = "a";
    final char middle = 'b';
    final String right = "c";
    final String replaced = "bb";

    @Rule
    public MockitoRule mockito = MockitoJUnit.rule();

    @Mock
    Scaling scaling; // fake
    @Mock
    ScalingRepeater repeater; // stub

    @Test
    public void equalsContract() {
        EqualsVerifier.forClass(Line.class). //
                withNonnullFields("line"). //
                verify();
    }

    /** 
     * @see ScalingRepeaterTest#shouldRepeatCharacter() for contract of stub.
     */
    @Test
    public void shouldScaleMiddleCharacterHorizontally() {
        when(repeater.repeat(middle, scaling)).thenReturn(replaced);

        Line line = new Line(left + middle + right);
        Line scaledLine = line.scaleHorizontal(repeater, scaling);

        assertEquals(left + replaced + right, scaledLine.toString());
    }
}

File: JavaScript/test/line_test.js
/* globals describe, it, beforeEach, Line */
'use strict';

var chai = require('chai');
var should = chai.should();
var sinon = require('sinon');

require('../src/line');

var left = 'a';
var middle = 'b';
var right = 'c';
var replaced = 'bb';

describe('Line', function() {
    var scaling; // fake object
    var repeat; // stub function

    beforeEach(function() {
        scaling = {};
    });

    it('should scale middle character horizontally', function() {
        repeat = sinon.stub().withArgs(middle, scaling).returns(replaced);

        var line = new Line(left + middle + right);
        var scaledLine = line.scaleHorizontal(repeat, scaling);

        scaledLine.toString().should.equal(left + replaced + right);
    });

});
